ide-support/revdeploylibraryemscripten.livecodescript

```diff
@@ -0,0 +1,231 @@
+            if tBrowserList is not empty then
+               put return after tBrowserList
+            end if
+            put queryRegistry("HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\" & tBrowser & backslash) & comma & \
+                  queryRegistry("HKEY_LOCAL_MACHINE\SOFTWARE\Clients\StartMenuInternet\" & tBrowser & "\shell\open\command\") after tBrowserList
+         end repeat
+         replace quote with empty in tBrowserList
+         break
+         
+      case "Linux"
+         -- For linux, we have a list of known browsers.
+         -- Check to see if the exe is in the path or installed in the user's home dir.
+         --
+         local tBrowserArray
+         put kLinuxBrowserList into tBrowserArray
+         split tBrowserArray by comma and "="
+         
+         set the itemDel to slash
+         repeat for each key tBrowser in tBrowserArray
+            get shell("which" && item 1 of tBrowserArray[tBrowser])
+            if it is not empty then
+               put it into tBrowserList[tBrowser]
+            else
+               local tUserBrowserPath
+               put $HOME & slash & tBrowserArray[tBrowser] into tUserBrowserPath
+               if there is a file tUserBrowserPath then
+                  put tUserBrowserPath into tBrowserList[tBrowser]
+               end if
+            end if
+         end repeat
+         set the itemDel to comma
+         
+         combine tBrowserList with return and comma
+         break
+         
+      default
+         put empty into tBrowserList
+         break
+   end switch
+   
+   return tBrowserList
+end __deployGetBrowserList
+
+private command __deployLaunchURLInBrowser pURL, pBrowser
+   if pBrowser is empty then
+      launch URL pURL
+      return the result
+   else
+      switch the platform
+         case "MacOS"
+            return shell("open -a" && __wrapInQuotes(pBrowser) && __wrapInQuotes(pURL))
+            
+         case "Win32"
+            local tHideConsoleWindows
+            put the hideConsoleWindows into tHideConsoleWindows
+            set the hideConsoleWindows to true
+            get shell(__wrapInQuotes(pBrowser) &&  __wrapInQuotes(pURL))
+            set the hideConsoleWindows to tHideConsoleWindows            
+            return it
+            
+         case "Linux"
+            return shell(__wrapInQuotes(pBrowser) &&  __wrapInQuotes(pURL))
+            
+         default
+            launch URL pURL
+            return the result
+      end switch      
+   end if
+end __deployLaunchURLInBrowser
+
+-- Callback from the HTTP deamon.
+-- In this case we just want to serve the requested file
+-- if it exists in the temp file space.
+--
+on deployEmscriptenNewHTTPRequest pSocketID, pRequest
+   local tPath   
+   put sHTTPFileSpace & slash & pRequest["resource"] into tPath
+   
+   if there is a file tPath then
+      httpdResponse pSocketID, 200, URL ("binfile:" & tPath)
+   else
+      httpdResponse pSocketID, 404
```

Comment:
@livecodemichael do we want to display the emrun callbacks anywhere? I think they could just go to the msg box as when remote debugger is in use put is routed over the socket to the IDE to put into the msg box so it's a bit similar.

Reply:
@montegoulding Yeah, I was thinking about that. Seems like a good idea.